eva_submission/evapro/populate_evapro.py

```diff
@@ -0,0 +1,454 @@
+    EvaSubmission, ProjectEvaSubmission, Analysis, AssemblySet, AccessionedAssembly, File, BrowsableFile, \
+    Platform, ExperimentType, Sample, SampleInFile
+from eva_submission.samples_checker import get_samples_from_vcf
+
+ena_ftp_file_prefix_path = "/ftp.sra.ebi.ac.uk/vol1"
+
+def get_ftp_path(filename, accession_id):
+    return f"{ena_ftp_file_prefix_path}/{accession_id[0:6]}/{accession_id}/{filename}"
+
+
+class EvaProjectLoader(AppLogger):
+
+    def __init__(self):
+        self.ena_project_finder = EnaProjectFinder()
+
+    def _evapro_engine(self):
+        pg_url, pg_user, pg_pass = get_metadata_creds_for_profile(cfg['maven']['environment'], cfg['maven']['settings_file'])
+        dbtype, host_url, port_and_db = urlsplit(pg_url).path.split(':')
+        port, db = port_and_db.split('/')
+        return create_engine(URL.create(
+            'postgresql+psycopg2',
+            username=pg_user,
+            password=pg_pass,
+            host=host_url.split('/')[-1],
+            database=db,
+            port=int(port)
+        ))
+
+    @cached_property
+    def eva_session(self):
+        session = Session(self._evapro_engine())
+        return session
+
+    def get_assembly_code_from_evapro(self, assembly):
+        query = select(AssemblySet.assembly_code)\
+            .join(AccessionedAssembly, AssemblySet.assembly_set_id==AccessionedAssembly.assembly_set_id)\
+            .where(AccessionedAssembly.assembly_accession==assembly)
+        rows = set(self.eva_session.execute(query).fetchall())
+        if len(rows) == 0:
+            return None
+        elif len(rows) > 1:
+            options = ', '.join([row for row, in rows])
+            raise ValueError(f'More than one possible code for assembly {assembly} found: {options}')
+        return rows[0][0]
+
+    def get_assembly_code(self, assembly, ncbi_api_key=None):
+        # TODO: deduplicate the code taken from ebi_eva_internal_pyutils.metadata_utils.py
+        assembly_code = self.get_assembly_code_from_evapro(assembly)
+        if not assembly_code:
+            assembly_name = get_ncbi_assembly_name_from_term(assembly, api_key=ncbi_api_key)
+            # If the assembly is a patch assembly ex: GRCh37.p8, drop the trailing patch i.e., just return grch37
+            if is_patch_assembly(assembly):
+                assembly_name = re.sub('\\.p[0-9]+$', '', assembly_name.lower())
+            assembly_code = re.sub('[^0-9a-zA-Z]+', '', assembly_name.lower())
+        return assembly_code
+
+    def get_platform_obj_from_evapro(self, platform):
+        query = select(Platform).where(Platform.platform == platform)
+        result = self.eva_session.execute(query).fetchone()
+        if result:
+            platform_obj = result.Platform
+            return platform_obj
+        return None
+
+    def eva_study_accession_from_project_accession_code(self):
+        # Is it to really necessary
```

Comment:
Honestly I'm kind of leaning towards the strategy of only porting what we know we'll need, then doing really thorough testing to make sure things aren't broken. Maybe it's slightly safer to port everything and gradually remove pieces, but the testing is a necessity either way...